### PR TITLE
allow sign in anonymous

### PR DESCRIPTION
**Pull request by nguyenhoaidanh**
signin in anonymous for all user in kyberswap

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -33,3 +33,7 @@ REACT_APP_LIMIT_ORDER_API_READ=https://limit-order.kyberswap.com/read-ks/api
 # Kyber DAO Apis
 REACT_APP_KYBER_DAO_STATS_API=https://kyberswap-dao-stats.kyberengineering.io
 REACT_APP_NOTIFICATION_IGNORE_TEMPLATE_IDS=2,29
+
+# oauth
+REACT_APP_ENV=development
```
Comment (viet-nv) [inline .env:38]: why do we need this?

File: src/hooks/useLogin.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  const requestingAnonymous = useRef(false)
+  useEffect(() => {
+    const signIn = async function signIn() {
+      try {
+        const ClientAppConfig = { clientId: OAUTH_CLIENT_ID }
+        KyberOauth2.initialize(ClientAppConfig)
+        if (!requestingAnonymous.current) {
+          requestingAnonymous.current = true
+          const data = await KyberOauth2.loginAnonymous()
+          console.log(data)
```
Comment (viet-nv) [inline src/hooks/useLogin.tsx:17]: console.log?

File: src/hooks/useLogin.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import KyberOauth2 from '@kybernetwork/oauth2'
+import { useEffect, useRef } from 'react'
+
+import { OAUTH_CLIENT_ID } from 'constants/env'
+
+const useLogin = () => {
+  // prevent spam because react 18 strict mode
+  const requestingAnonymous = useRef(false)
```
Comment (viet-nv) [inline src/hooks/useLogin.tsx:8]: I didnt get your idea. why isn't it a loading state?

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -75,6 +75,8 @@ jobs:
         uses: actions/setup-node@v1
         with:
           node-version: 16.18.1
+          registry-url: 'https://npm.pkg.github.com'
+          scope: '@kybernetwork'
```
[the remark this answers is not in the document]
Comment (nguyenhoaidanh) [reply .github/workflows/ci.yaml:79]: currently, sdk kybernetwork/oauth2 is private in our kybernetwork organization github. we are considerring public or private it.

this config for CI can download this private package

File: src/hooks/useLogin.tsx (hunk 3 above)
Comment (namgold) [reply src/hooks/useLogin.tsx:8]: Isn't empty array dependencies solved this? Why do you need an additional ref

Comment (nguyenhoaidanh) [reply src/hooks/useLogin.tsx]: ah. nice, move it into index.tsx.

File: .env (hunk 1 above)
Comment (nguyenhoaidanh) [reply .env:38]: to select which mode sdk can use, for test only. when we officially public this SDK. I will remove it

**Review by viet-nv — DISMISSED** [review]

File: .env (hunk 1 above)
Comment (viet-nv) [reply .env:38]: okay, you can add a TODO on it

**Review by namgold — APPROVED** [review]

**Pull request merged**